### PR TITLE
Update FileSaver to include a module definition

### PR DESCRIPTION
**Pull request by elliott-davis**
Improvement to existing type definition.
- There is no documentation updates for this change. It was simply missing a module declaration.

### CODE REVIEW

**Comment by chris-rock** [comment]
Great catch @elliott-davis This helps to ease the process with webpack

**Comment by cyrilschumacher** [comment]
👍

File: FileSaver/FileSaver.d.ts [hunk 1]
```diff
@@ -30,4 +30,8 @@ interface FileSaver {
     ): void
 }
 
-declare var saveAs: FileSaver;
```
Comment (mhegazy) [inline FileSaver/FileSaver.d.ts]: The package documentation in https://github.com/eligrey/FileSaver.js refers to a global `saveAs`.

```ts
var blob = new Blob(["Hello, world!"], {type: "text/plain;charset=utf-8"});
saveAs(blob, "hello world.txt");
```

when used as a module however, `saveAs` would be a property on the module, see https://github.com/eligrey/FileSaver.js/blob/master/FileSaver.js#L183

so it should be:

```ts
declare var saveAs: FileSaver;

declare module "file-saver" {
    var fileSaver: { saveAs: typeof saveAs };
    export = fileSaver
}
```

**Comment by cyrilschumacher** [comment]
:+1:

would it be possible to do a test for this feature?

**Comment by elliott-davis** [comment]
I think this just makes the import line work correctly - I'm not 100% how I would write a test for the module. I'd be happy to patch it up if anyone has suggestions on how to test it.

**Comment by mhegazy** [comment]
in https://github.com/elliott-davis/DefinitelyTyped/blob/294a09c342c2a3e91a8a43abd200f99d0fd377f6/FileSaver/FileSaver-tests.ts add:

```ts
import {saveAs as importedSaveAs} from "file-saver";
function testImportedSaveAs() {
    var data: Blob = new Blob(["Hello, world!"], {type: "text/plain;charset=utf-8"});
    var filename: string = 'hello world.txt';
    var disableAutoBOM = true;

    importedSaveAs(data, filename, disableAutoBOM);
}
```

**Comment by cyrilschumacher** [comment]
:+1: Thanks @elliott-davis and @mhegazy for the test

**Pull request merged**